src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+            self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+            self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.INSTALLED)
+
+        truncate_end_time = time.time()
+        truncate_performance_time = truncate_end_time - truncate_start_time
+        truncate_performance_time_formatted = self.__convert_performance_time_to_date_time_format(truncate_performance_time)
+
+        self.__remove_tmp_file_reset_stdout()  # remove and reset tmp file for storing sys.stout()
+
+        self.runtime.status_handler.composite_logger.log_debug('no_truncate_performance_time_formatted' + no_truncate_performance_time_formatted)
+        self.runtime.status_handler.composite_logger.log_debug('truncate_performance_time_formatted' + truncate_performance_time_formatted)
+        self.assertTrue((truncate_performance_time - no_truncate_performance_time) < 30)
+
+    # Setup functions for testing
+    def __assert_packages_map(self, substatus_file_data, patch_summary, patch_count, expected_patch_id, expected_patch_value, classification):
+        message = json.loads(substatus_file_data['formattedMessage']['message'])
+        self.assertEqual(substatus_file_data["name"], patch_summary)
+        self.assertEqual(len(message['patches']), patch_count)
+        self.assertEqual(message['patches'][0]['patchId'], expected_patch_id)
+        self.assertEqual(message['patches'][0]['name'], 'python-samba0')
+        self.assertEqual(message['patches'][0], expected_patch_value)
+        self.assertEqual(message['patches'][0]['classifications'], [classification])
+
+    def __assert_installation_set_packages_methods(self, substatus_file_data, patch_summary, patch_count):
+        message = json.loads(substatus_file_data["formattedMessage"]["message"])
+        self.assertEqual(substatus_file_data["name"], patch_summary)
+        self.assertEqual(len(message["patches"]), patch_count)
+        self.assertEqual(message["patches"][0]["name"], "python-samba0")
+        self.assertTrue('Critical' in str(message["patches"][0]["classifications"]))
+        self.assertEqual(message["patches"][1]["name"], "python-samba1")
+        self.assertEqual('Critical', str(message["patches"][1]["classifications"][0]))
+        self.assertEqual('Installed', str(message["patches"][1]["patchInstallationState"]))
+        self.assertEqual(message["patches"][2]["name"], "python-samba2")
+        self.assertEqual('python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04', str(message["patches"][0]["patchId"]))
+        self.assertTrue('Critical' in str(message["patches"][2]["classifications"]))
+
+        # Clean up complete.status files
+        self.runtime.env_layer.file_system.delete_files_from_dir(self.runtime.status_handler.status_file_path, '*.complete.status')
+
+    def __assert_no_truncation_status_file(self, patch_summary, status, patch_count):
```

Comment:
Since this is validating no truncation occurred, the important assert should be that the status file size before and after truncation function is the same.

Or better yet, complete status file size and status file size is the same

Reply:
self.assertEqual(len(json.dumps(substatus_file_data)), len(json.dumps(complete_status_file_data))) added. done